library/java/src/io/envoyproxy/envoymobile/engine/JniLibrary.java

```diff
@@ -69,6 +69,17 @@ public static void load() {
    */
   protected static native int sendHeaders(long stream, byte[][] headers, boolean endStream);
 
+  /**
+   * Send data over an open HTTP stream. This method can be invoked multiple
+   * times.
+   *
+   * @param stream,    the stream to send data over.
+   * @param data,      the data to send.
+   * @param endStream, supplies whether this is the last data in the stream.
+   * @return int, the resulting status of the operation.
+   */
+  protected static native int sendData(long stream, byte[] data, boolean endStream);
```

Comment:
Should we update the other method's documentation to call out that we need to direct buffer?